terminal/terminal.go

```diff
@@ -150,6 +167,34 @@ func (t *Term) Println(prefix, str string) {
 	fmt.Fprintf(t.stdout, "%s%s\n", prefix, str)
 }
 
+// SubstitutePath substitues directory to soruce file.
+//
+// Ensures that only directory is substitued, for example:
+// substitute from `/dir/subdir`, substitute to `/new`
+// for file path `/dir/subdir/file` will return file path `/new/file`.
+// for file path `/dir/subdir-2/file` substitution will not be applied.
+//
+// If more than one substitution rule is defiend, the rules are applied
+// in the order they are defined, first rule that matches is used for
+// substitution.
+func (t *Term) SubstitutePath(path string) string {
+	separator := string(os.PathSeparator)
```

Comment:
This will not work on windows. See path/filepath package for functions to manipulate file paths. I would also add new test to verify your changes actually work now and in the future.